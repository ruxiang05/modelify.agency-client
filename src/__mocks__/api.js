const jobMock = {
  status: 'in progress',
  _id: '5c97b9cd0328e6b2aa76a361',
  model: '5c686eb24cb1b33456b0be59',
  agent: '5c686e904cb1b33456b0be58',
  title: 'Test',
  date: '2019-05-26T00:00:00.000Z',
  address: 'address',
  pay: 100,
  description: 'some description',
};

const api = {
  jobs: {
    getJobs: jest.fn(() => Promise.resolve({
      jobs: [jobMock],
    })),
    acceptJob: jest.fn(() => Promise.resolve({
      job: jobMock,
    })),
    completeJob: jest.fn(() => Promise.resolve({
      job: {
        ...jobMock,
        status: 'complete',
      },
    })),
    declineJob: jest.fn(() => Promise.resolve({
      job: {
        ...jobMock,
        status: 'declined',
      },
    })),
    addJob: jest.fn(() => Promise.resolve()),
  },
  googleCalendar: {
    addEvent: jest.fn(() => Promise.resolve()),
    authorize: jest.fn(() => Promise.resolve({ message: 'success', token: 'test' })),
    getAuthURL: jest.fn(() => Promise.resolve({ redirect: '/redirect' })),
  },
  agents: {
    getModels: jest.fn(() => Promise.resolve({
      agent: {
        agentInfo: {
          models: [
            {
              _id: '5ca24d217736960017cae87c',
              agentInfo: { models: [] },
              email: 'jane@doe.com',
              modelInfo: {
                address: '',
                characteristics: {
                  eyes: '',
                  hair: '',
                  skin: '',
                  dateOfBirth: '',
                },
                measurements: {
                  chest: '',
                  height: '',
                  hips: '',
                  waist: '',
                  weight: '',
                },
                name: 'Jane Doe',
                phoneNumber: '07423360809',
                role: 'model',
              },
            },
          ],
        },
      },
    })),
    addModel: jest.fn(() => Promise.resolve()),
  },
  chats: {
    getMessages: jest.fn(() => Promise.resolve({
      messages: [
        {
          message: 'hi',
          user: '1',
        },
        {
          message: 'hello',
          user: '2',
        },
      ],
    })),
    getChats: jest.fn(() => Promise.resolve({
      chats: [
        {
          id: '2t3457345',
          job: {},
          messages: [],
          users: ['1', '2'],
        },
      ],
    })),
  },
  users: {
    signup: jest.fn(() => Promise.resolve({ ok: true })),
    login: jest.fn(() => Promise.resolve({ token: 'abc' })),
    editProfile: jest.fn(() => Promise.resolve({ token: 'abc' })),
  },
};

module.exports = api;
