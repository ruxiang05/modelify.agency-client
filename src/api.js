const url = process.env.REACT_APP_API_URL;

const api = {
  users: {
    signup: data => fetch(`${url}/users/signup`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
      },
    }),
    login: data => fetch(`${url}/users/login`, {
      method: 'POST',
      body: JSON.stringify(data),
      headers: {
        'Content-Type': 'application/json',
      },
    }),
    editProfile: (token, data) => fetch(`${url}/users/edit`, {
      method: 'PUT',
      body: JSON.stringify(data),
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  agents: {
    getModels: token => fetch(`${url}/agents/`, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    addModel: (token, data) => fetch(`${url}/agents/add-model`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
  jobs: {
    addJob: (token, data) => fetch(`${url}/jobs/create`, {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'content-type': 'application/json',
      },
      body: JSON.stringify(data),
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
    getJobs: token => fetch(`${url}/jobs/all`, {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    })
      .then((res, err) => {
        if (err) {
          return err;
        }
        return res.json();
      })
      .catch(err => err),
  },
};

export default api;
