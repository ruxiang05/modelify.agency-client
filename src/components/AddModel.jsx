import React from 'react';
import PropTypes from 'prop-types';
import FormInput from './FormInput';
import PageHeader from './PageHeader';
import api from '../api';
import { getToken } from '../auth';

class AddModel extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    const { value } = event.target;
    this.setState(prevState => ({ ...prevState, email: value }));
  }

  handleSubmit(event) {
    const { email } = this.state;
    const { history } = this.props;
    event.preventDefault();
    api.agents.addModel(getToken(), { model: email });
    history.push('/models');
  }

  render() {
    const { email } = this.state;
    const { history } = this.props;
    return (
      <React.Fragment>
        <PageHeader title="Add model" backButton history={history} />
        <div className="page">
          <form onSubmit={this.handleSubmit}>
            <FormInput
              type="text"
              title="Add model"
              name="add-model"
              value={email}
              placeholder="Model email"
              handleChange={this.handleChange}
            />
            <div className="form-submit">
              <input type="submit" value="Submit" />
            </div>
          </form>
        </div>
      </React.Fragment>
    );
  }
}

AddModel.propTypes = {
  history: PropTypes.shape({}).isRequired,
};

export default AddModel;
