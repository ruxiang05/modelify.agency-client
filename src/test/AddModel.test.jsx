/* Uses methods and/or components from react, history and enzyme */
import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import AddModel from '../components/AddModel';
import api from '../api';

jest.mock('../api.js');

let wrapper;

const history = createBrowserHistory({
  basename: '/',
  keyLength: 0,
});

const formSubmit = {
  email: 'email@email.com',
};

beforeEach(() => {
  wrapper = shallow(<AddModel history={history} />);
});

describe('AddModel', () => {
  it('should render AddModel', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle input change', async () => {
    const event = {
      target: {
        value: 'email@email.com',
      },
    };
    wrapper.instance().handleChange(event);
    await wrapper.update();
    expect(wrapper.state().email).toBe('email@email.com');
  });

  it('should handle form submit', async () => {
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState({ ...formSubmit });
    await wrapper.update();
    wrapper.instance().handleSubmit(event);
    expect(api.agents.addModel).toHaveBeenCalled();
    expect(history.location.pathname).toBe('/models');
  });
});
