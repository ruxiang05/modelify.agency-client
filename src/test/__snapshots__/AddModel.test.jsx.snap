// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddModel should render AddModel 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddModel
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader
          backButton={true}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          title="Add model"
        />,
        <div
          className="page"
        >
          <form
            onSubmit={[Function]}
          >
            <FormInput
              handleChange={[Function]}
              name="add-model"
              placeholder="Model email"
              title="Add model"
              type="text"
              value=""
            />
            <div
              className="form-submit"
            >
              <input
                type="submit"
                value="Submit"
              />
            </div>
          </form>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "action": undefined,
          "actionAltText": undefined,
          "actionIcon": undefined,
          "backButton": true,
          "goBackAction": undefined,
          "history": Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          },
          "title": "Add model",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
          >
            <FormInput
              handleChange={[Function]}
              name="add-model"
              placeholder="Model email"
              title="Add model"
              type="text"
              value=""
            />
            <div
              className="form-submit"
            >
              <input
                type="submit"
                value="Submit"
              />
            </div>
          </form>,
          "className": "page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FormInput
                handleChange={[Function]}
                name="add-model"
                placeholder="Model email"
                title="Add model"
                type="text"
                value=""
              />,
              <div
                className="form-submit"
              >
                <input
                  type="submit"
                  value="Submit"
                />
              </div>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "handleChange": [Function],
                "name": "add-model",
                "placeholder": "Model email",
                "title": "Add model",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  type="submit"
                  value="Submit"
                />,
                "className": "form-submit",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "submit",
                  "value": "Submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader
            backButton={true}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            title="Add model"
          />,
          <div
            className="page"
          >
            <form
              onSubmit={[Function]}
            >
              <FormInput
                handleChange={[Function]}
                name="add-model"
                placeholder="Model email"
                title="Add model"
                type="text"
                value=""
              />
              <div
                className="form-submit"
              >
                <input
                  type="submit"
                  value="Submit"
                />
              </div>
            </form>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "action": undefined,
            "actionAltText": undefined,
            "actionIcon": undefined,
            "backButton": true,
            "goBackAction": undefined,
            "history": Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            },
            "title": "Add model",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
            >
              <FormInput
                handleChange={[Function]}
                name="add-model"
                placeholder="Model email"
                title="Add model"
                type="text"
                value=""
              />
              <div
                className="form-submit"
              >
                <input
                  type="submit"
                  value="Submit"
                />
              </div>
            </form>,
            "className": "page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FormInput
                  handleChange={[Function]}
                  name="add-model"
                  placeholder="Model email"
                  title="Add model"
                  type="text"
                  value=""
                />,
                <div
                  className="form-submit"
                >
                  <input
                    type="submit"
                    value="Submit"
                  />
                </div>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": [Function],
                  "name": "add-model",
                  "placeholder": "Model email",
                  "title": "Add model",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    type="submit"
                    value="Submit"
                  />,
                  "className": "form-submit",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "type": "submit",
                    "value": "Submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "div",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
